extensions-contrib/ganglia-emitter/src/main/java/org/apache/druid/emitter/ganglia/GangliaEmitter.java

```diff
@@ -0,0 +1,250 @@
+      String fullName = Joiner.on(config.getSeparator())
+                              .join(nameBuilder.build());
+      fullName = StringUtils.replaceChar(fullName, DRUID_METRIC_SEPARATOR, config.getSeparator());
+      fullName = STATSD_SEPARATOR.matcher(fullName).replaceAll(config.getSeparator());
+      fullName = BLANK.matcher(fullName).replaceAll(config.getBlankHolder());
+      GangliaEvent gangliaEvent = new GangliaEvent(fullName, metricEvent.getValue(), StringUtils.replaceChar(service, DRUID_METRIC_SEPARATOR, config.getSeparator()));
+      try {
+        final boolean successful = eventsQueue.offer(
+            gangliaEvent,
+            config.getEmitWaitTime(),
+            TimeUnit.MILLISECONDS
+        );
+        if (!successful) {
+          if (countLostEvents.getAndIncrement() % 1000 == 0) {
+            log.error(
+                "Lost total of [%s] events because of emitter queue is full. Please increase the capacity or/and the consumer frequency",
+                countLostEvents.get()
+            );
+          }
+        }
+      }
+      catch (InterruptedException e) {
+        log.error(e, "Emitter metric interrupted with message [%s]", e.getMessage());
+        Thread.currentThread().interrupt();
+      }
+    }
+  }
+
+  @Override
+  public void flush() throws IOException
+  {
+  }
+
+  @Override
+  public void close() throws IOException
+  {
+    log.info("Closing Ganglia Emitter");
+    ganglia.close();
+  }
+
+  private Map<String, GangliaMetric> readMap(ObjectMapper mapper, String dimensionMapPath)
+  {
+    try {
+      InputStream is;
+      if (Strings.isNullOrEmpty(dimensionMapPath)) {
+        log.info("Using default ganglia metric dimension and types");
+        is = this.getClass().getClassLoader().getResourceAsStream("defaultMetricDimensions.json");
+      } else {
+        log.info("Using ganglia metric dimensions at types at [%s]", dimensionMapPath);
+        is = new FileInputStream(new File(dimensionMapPath));
+      }
+      return mapper.readerFor(new TypeReference<Map<String, GangliaMetric>>()
+      {
+      }).readValue(is);
+    }
+    catch (IOException e) {
+      throw new ISE(e, "Failed to parse ganglia metric dimensions and types");
+    }
+  }
+
+  private class ConsumerRunnable implements Runnable
+  {
+    @Override
+    public void run()
+    {
+      while (eventsQueue.size() > 0 && !exec.isShutdown()) {
+        try {
+          final GangliaEvent gangliaEvent = eventsQueue.poll(
+              config.getWaitForEventTime(),
+              TimeUnit.MILLISECONDS
+          );
+          if (gangliaEvent != null) {
+            try {
+              ganglia.announce(
+                  gangliaEvent.getName(),
+                  gangliaEvent.getValue().longValue(),
+                  gangliaEvent.getService()
+              );
+            }
+            catch (GangliaException e) {
+              log.error(e, "Dropping event [%s]. Unable to send to Ganglia.", gangliaEvent);
```

Comment:
Would this generate an excessive number of messages (with ERROR severity) whenever there's an issue, maybe for example a temporary network glitch preventing communication with Ganglia? Consider logging the first one (at info level) and then suppressing logging on the rest of them.